### PR TITLE
Classifying more UNKNOWN_ERRORs

### PR DESCRIPTION
**Pull request by melissaahn**
### Summary
The error code `UNKNOWN_ERROR` appears frequently in our telemetry. We have a large amount of client exceptions going through the `clientExceptionFromException` method with an `UNKNOWN_ERROR` code, and there are also many places where `ClientException`s are being explicitly created with the code. 
After looking at the telemetry and doing some code searches in our codebase, I added some more specific error codes which will be used in the common and broker libraries. Any suggestions on name changes are welcomed.

### Related PRs

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public static BaseException exceptionFromAuthorizationResult(@NonNull  final Aut
                     methodTag,
                     "AuthorizationErrorResponse is not set"
             );
-            return new ClientException(ClientException.UNKNOWN_ERROR, "Authorization failed with unknown error. Authorization Status: " +authorizationResult.getAuthorizationStatus());
+            return new ClientException(ClientException.AUTHORIZATION_RESULT_ERROR, "Authorization failed with unknown error. Authorization Status: " +authorizationResult.getAuthorizationStatus());
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java]: We can probably rename the error code here to `AUTHORIZATION_RESULT_NULL_ERROR_RESPONSE`

**Review by shahzaibj — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java (hunk 1 above)
Comment (rpdome) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java]: +1 (and also improve the error message accordingly?)

**Review by rpdome — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java (hunk 1 above)
Comment (melissaahn) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java]: Addressed here: 96ab98a6d06c248fb699749acea717d36f61c015

**Comment by melissaahn** [comment]
I've tagged this PR with "skip-consumers-check" because there is one MSAL unit test which is expected to fail. See this PR: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2147

**Pull request merged**